libraries/exoplayer_hls/src/main/java/androidx/media3/exoplayer/hls/playlist/HlsPlaylistParser.java

```diff
@@ -374,7 +411,27 @@ private static HlsMultivariantPlaylist parseMultivariantPlaylist(
         int roleFlags = isIFrameOnlyVariant ? C.ROLE_FLAG_TRICK_PLAY : 0;
         int peakBitrate = parseIntAttr(line, REGEX_BANDWIDTH);
         int averageBitrate = parseOptionalIntAttr(line, REGEX_AVERAGE_BANDWIDTH, -1);
+        String videoRange = parseOptionalStringAttr(line, REGEX_VIDEO_RANGE, variableDefinitions);
         String codecs = parseOptionalStringAttr(line, REGEX_CODECS, variableDefinitions);
+        String supplementalCodecsStrings =
+            parseOptionalStringAttr(line, REGEX_SUPPLEMENTAL_CODECS, variableDefinitions);
+        String supplementalCodecs = null;
+        String supplementalProfiles = null;  // i.e. Compatibility brand
+        if (supplementalCodecsStrings != null) {
+          String[] supplementalCodecsString = Util.split(supplementalCodecsStrings, ",");
+          if (!supplementalCodecsString[0].isEmpty()) {
+            // TODO: Support more than one element
+            String[] codecsAndProfiles = Util.split(supplementalCodecsString[0], "/");
+            supplementalCodecs = codecsAndProfiles[0];
+            if (codecsAndProfiles.length > 1) {
+              supplementalProfiles = codecsAndProfiles[1];
+            }
+          }
+        }
+        if (isDolbyVisionFormat(videoRange, codecs, supplementalCodecs, supplementalProfiles)) {
```

Comment:
There is a problem here. Because I move this part of logic from HlsMediaPeriod.java to HlsPlaylistParser.java and the "codecs" is a string that includes both audio and video codec name, I can't update the codecs directly. I need to refine this part of code again.

Reply:
I think this can be resolved because we now have the improved logic that re-adds the other codecs.